complete/helpers.c

```diff
@@ -232,7 +232,21 @@ int mutt_command_complete(struct CompletionData *cd, struct Buffer *buf, int pos
   }
   else if (buf_startswith(buf, "exec"))
   {
-    const enum MenuType mtype = menu_get_current_type();
+    enum MenuType mtype = MENU_GENERIC;
+    if (cdata)
+    {
+      struct FileCompletionData *fcd = cdata;
+      struct MuttWindow *win = fcd->win;
+      if (win && win->wdata)
+      {
+        struct Menu *menu = win->wdata;
+        mtype = menu->type;
+      }
+    }
+    else
+    {
+      mtype = menu_get_current_type();
+    }
```

Comment:
I doesn't hurt but does this ever make sense?

Reply:
I'm fairly certain it can't get reached, but it won't cause any harm.